sbt-plugin/src/main/scala/org/scalajs/sbtplugin/ScalaJSConfigJSONProtocol.scala

```diff
@@ -0,0 +1,214 @@
+      }
+      builder.endObject()
+    }
+
+    override def read[J](jsOpt: Option[J], unbuilder: Unbuilder[J]): RuntimeClassNameMapper = jsOpt match {
+      case Some(js) =>
+        unbuilder.beginObject(js)
+        val mapper = unbuilder.readField[String]("kind") match {
+          case "KeepAll" => KeepAll
+          case "DiscardAll" => DiscardAll
+          case "RegexReplace" =>
+            val pattern = unbuilder.readField[String]("pattern")
+            val flags = unbuilder.readField[Int]("flags")
+            val replacement = unbuilder.readField[String]("replacement")
+            RegexReplace(pattern, flags, replacement)(Pattern.compile(pattern))
+          case "AndThen" =>
+            val first = read(unbuilder.lookupField("first"), unbuilder)
+            val second = read(unbuilder.lookupField("second"), unbuilder)
+            AndThen(first, second)
+        }
+        unbuilder.endObject()
+        mapper
+
+      case None => deserializationError("Expected JsObject but found None")
+    }
+  }
+
+
+  type SemanticsJSON =
+    CheckedBehavior :*: CheckedBehavior :*: CheckedBehavior :*: Boolean :*:
+    Boolean :*: Semantics.RuntimeClassNameMapper :*: LNil
+
+  implicit val semanticsFormat = LList.iso[Semantics, SemanticsJSON](
+    { sem: Semantics =>
+      ("asInstanceOfs", sem.asInstanceOfs) :*:
+      ("arrayIndexOutOfBounds", sem.arrayIndexOutOfBounds) :*:
+      ("moduleInit", sem.moduleInit) :*:
+      ("strictFloats", sem.strictFloats) :*:
+      ("productionMode", sem.productionMode) :*:
+      ("runtimeClassNameMapper", sem.runtimeClassNameMapper) :*:
+      LNil
+    },
+    { case (_, asInstanceOfs) :*: (_, arrayIndexOutOfBounds) :*: (_, moduleInit) :*:
+           (_, strictFloats) :*: (_, productionMode) :*:
+           (_, runtimeClassNameMapper) :*: LNil =>
+        Semantics.Defaults
+          .withAsInstanceOfs(asInstanceOfs)
+          .withArrayIndexOutOfBounds(arrayIndexOutOfBounds)
+          .withModuleInit(moduleInit)
+          .withStrictFloats(strictFloats)
+          .withProductionMode(productionMode)
+          .withRuntimeClassNameMapper(runtimeClassNameMapper)
+    }
+  )
+
+  type StandardConfigJSON =
+    ModuleKind :*: ESFeatures :*: Boolean :*: Boolean :*:
+    Boolean :*: Boolean :*: Option[URI] :*: Boolean :*: Boolean :*:
+    Boolean :*: Semantics :*: LNil
+
+  implicit val standardConfigFormat = LList.iso[StandardConfig, StandardConfigJSON](
+    { conf: StandardConfig =>
+      ("moduleKind", conf.moduleKind) :*:
+      ("esFeatures", conf.esFeatures) :*:
+      ("checkIR", conf.checkIR) :*:
+      ("optimizer", conf.optimizer) :*:
+      ("parallel", conf.parallel) :*:
+      ("sourceMap", conf.sourceMap) :*:
+      ("relativizeSourceMapBase", conf.relativizeSourceMapBase) :*:
+      ("closureCompilerIfAvailable", conf.closureCompilerIfAvailable) :*:
+      ("prettyPrint", conf.prettyPrint) :*:
+      ("batchMode", conf.batchMode) :*:
+      ("semantics", conf.semantics) :*:
```

Comment:
There's a bug in sjson-new where if I place the `semantics` object in the beginning, the deserializer crashes. I haven't been able to to a minimal bug repro of this, but placing the object at the end seems to solve the problem :man_shrugging: